### ISSUE TITLE
TPREFETCH_ASYNC gain only stabilizes after many prefetch issues; issues spike ~200-300 us before that

### ISSUE THREAD

**Issue by zhangqi-chen** [hw-native-sys/pto-isa#307]
## Background

In pypto-lib, the L2 SDMA CMO warm (`pl.prefetch.make_context` / `async_prefetch`) on a2a3 has a **performance** problem. **The gain from prefetch only settles after many prefetch issues have run. Before that, some rounds spike.** Right after process start, a fraction of `TPREFETCH_ASYNC` issues take ~200–300 us instead of ~24 us, and the tasks that depend on them finish late. The spikes fade as more prefetch issues run.

In DeepSeek V4 Flash decode, the rate of decode_fwd rounds with spikes falls from 2.3% at warmup 20 to 1.3% at warmup 100 and 0.4% at warmup 500. Only with long warmups does the prefetch gain become measurable.

Reproduce (single card, real device only; takes ~15 s):

    PYPTO_BENCH=1 PYPTO_BENCH_RAW=1 PYPTO_BENCH_WARMUP=20 PYPTO_BENCH_ROUNDS=300         python tprefetch_sqe_stall_min.py -p a2a3 -d 0

The repro case is **not committed**. Save it at the pypto-lib root; it only needs the `golden/` harness from pypto-lib `main` (d6f29200). Each dispatch runs 16 single-block `CORE_GROUP` warm scopes (64 MiB BF16 + 32 MiB INT8), each followed by a small task that depends on it.

<details>
<summary>Reproduction case — <code>tprefetch_sqe_stall_min.py</code> (save at the pypto-lib root)</summary>

```python
"""TPREFETCH_ASYNC issue stalls during the first few thousand posts of a process.

Each dispatch runs LAYERS single-block CORE_GROUP scopes. Each scope warms a 64 MiB
BF16 region and a 32 MiB INT8 region, and a small task depends on it, so a stalled
warm issue lengthens the dispatch directly. Run from the pypto-lib root:

    PYPTO_BENCH=1 PYPTO_BENCH_RAW=1 PYPTO_BENCH_WARMUP=20 PYPTO_BENCH_ROUNDS=300 \\
        python tprefetch_sqe_stall_min.py -p a2a3 -d 0
"""

import pypto.language as pl

LAYERS = 16
N_A = 8 * 1024 * 4096  # 64 MiB of BF16
N_B = 4096 * 8192      # 32 MiB of INT8
SEED = 64


@pl.jit
def prefetch_stall(
    w_a: pl.Tensor[[LAYERS * N_A], pl.BF16],
    w_b: pl.Tensor[[LAYERS * N_B], pl.INT8],
    seed: pl.Tensor[[1, SEED], pl.FP32],
    marker: pl.Out[pl.Tensor[[LAYERS, SEED], pl.FP32]],
):
    prev_tid = pl.system.task_dummy(deps=[])
    for layer in pl.unroll(LAYERS):
        a_flat = w_a[layer * N_A : (layer + 1) * N_A]
        b_flat = w_b[layer * N_B : (layer + 1) * N_B]
        with pl.at(level=pl.Level.CORE_GROUP, name_hint="warm", deps=[prev_tid]) as warm_tid:
            warm_ctx = pl.prefetch.make_context()
            pl.prefetch.async_prefetch(a_flat, warm_ctx)
            pl.prefetch.async_prefetch(b_flat, warm_ctx)
        with pl.at(level=pl.Level.CORE_GROUP, name_hint="after_warm", deps=[warm_tid]) as done_tid:
            marker[layer : layer + 1, 0:SEED] = pl.mul(seed[0:1, 0:SEED], 1.0)
        prev_tid = done_tid
    return marker


def golden(tensors):
    tensors["marker"][:] = tensors["seed"].expand(LAYERS, SEED)


if __name__ == "__main__":
    import argparse

    import torch
    from golden import TensorSpec, run

    parser = argparse.ArgumentParser()
    parser.add_argument("-p", "--platform", default="a2a3")
    parser.add_argument("-d", "--device", type=int, default=0)
    args = parser.parse_args()
    specs = [
        TensorSpec("w_a", [LAYERS * N_A], torch.bfloat16, init_value=lambda: torch.ones(LAYERS * N_A, dtype=torch.bfloat16)),
        TensorSpec("w_b", [LAYERS * N_B], torch.int8, init_value=lambda: torch.ones(LAYERS * N_B, dtype=torch.int8)),
        TensorSpec("seed", [1, SEED], torch.float32, init_value=torch.randn),
        TensorSpec("marker", [LAYERS, SEED], torch.float32),
    ]
    result = run(fn=prefetch_stall, specs=specs, golden_fn=golden,
                 config={"platform": args.platform, "device_id": args.device}, rtol=1e-6, atol=1e-6)
    if not result.passed:
        raise SystemExit(result.error)
```

</details>

Observed per-round wall time (`PYPTO_BENCH_RAW`, 300 rounds). A "slow" round is one more than 150 us above the median: median 355 us, mean 439 us, max 1218 us. There were 53 slow rounds, 52 of them in rounds 0–74 (16 / 19 / 17 per 25 rounds, then 0).

Reproduction environment:

| Component | Version |
|---|---|
| pypto-lib | `d6f29200` (main) |
| pypto | `706285c6` (main) |
| simpler | `39ce891d` (detached) |
| ptoas | `0.61` |
| pto-isa | `5a4f74cb` (detached) |
| CANN | `9.0.0` |

All components are on their pins.

Diagnosis: **pto-isa**. The stalls are in the `TPREFETCH_ASYNC` issue path.

## Expected

After runtime start-up, prefetch issue latency should be steady from the first dispatch, with no long warmup needed.

## Experiments

These are raw experiment records. We draw no conclusions from them.

1. **SQE block size** (`kSingleSqeBlockBytes` in `include/pto/comm/async/sdma/TPrefetchAsyncImpl.hpp`; set to 64 MiB since 7ddaa823, previously `kDefaultSdmaBlockBytes` = 1 MiB). Repro above, warmup 20, 300 rounds:

   | block size | median us | max us | slow rounds |
   |---|---:|---:|---:|
   | 64 MiB (pin) | 358 | 3133 | 55, all in rounds 0–82 |
   | 8 MiB | – | – | 9, all in rounds 0–8 |
   | 1 MiB | 570 | 615 | 0 |

   At 64 MiB an issue costs ~24 us; at 1 MiB it costs ~39 us.
2. **Warm posts per dispatch.** With 1, 4, or 16 warm scopes per dispatch, and with a 64 MiB warm only, a 32 MiB warm only, or both, the slow rounds stopped after roughly 3300–4900 warm posts in total. With 8 MiB blocks they stopped after ~3.7x fewer posts.
3. **Tiny warms first.** Running many 4 KB warms before the 64 MiB ones changed the slow rounds per 25 rounds from 18 / 16 / 14 / 5 (control) to 16 / 2.
4. **Longer simpler start-up SDMA warmup**, extended to every (AIV, channel) pair: the spike rate did not change (2.5% at warmup 20).
5. **Init-time burn.** An extra 8192 `TPREFETCH_ASYNC` posts on channel group 0 at init, from a single core: 43 slow rounds vs 53 without it.
6. **Channel selection.** These warms select `channelGroupIdx = get_block_idx()`, which is 0 for a single-block `CORE_GROUP` scope.
7. **Multi-card.** In DeepSeek V4 Flash decode at ep8, a stall on one rank also delays the other ranks through the MoE all-to-all.
